### DISCUSSION TITLE
Exception: validation.pattern.invalid.authenticationRequest.identifier

### DISCUSSION THREAD

**Discussion by Viforel** [ig-python/ig-markets-api-python-library#294] [Q&A]
Hello. I bumped my head left and right on my installation process but every time I try crating a new session, I get the following error:
> Exception: validation.pattern.invalid.authenticationRequest.identifier

1. I did a pip install including the dependencies in PyCharm, updated the config file with the proper **_credentials_** and rename it as per the instructions: **_trading_ig_config.py_**

A. If I am changing the API key for instance I am getting an expected error on the incorrect API so the configuration file variables are clearly passed into the main python file (validate by using the debugger).

B. Using the IG companion (https://labs.ig.com/sample-apps/api-companion/index.html) with the credentials added in the config file I can connect to the account:

2. Here is the super basic main.py I am using:

> from trading_ig.rest import IGService
> from trading_ig.config import config
> ig_service = IGService(config.username, config.password, config.api_key, config.acc_type, acc_number=config.acc_number)
> ig_service.create_session()

I am sure I am overlooking a basic thing but can't find what. I will be very helpful for any suggestions.

Thousand thanks,
Zoran

**Comment by LordNadimi** [comment]
Hey buddy, I am new here as well, but I am passed this part :) Using pycharm same as you.

I was looking at my code and to double check yours, as mine is working, but found no error. But after a google search on your issue:

So I ran a couple of tests using my LIVE credentials to share with helpdesk.au@.ig.com to demonstrate that if you use the right credentials, you actually get a different error.. i.e. you get a 403 error:

- of '{"errorCode":"error.security.invalid-details"}' if you use the wrong password

- of '{"errorCode":"validation.pattern.invalid.authenticationRequest.identifier"}'  if you use the wrong username and password

- of  '{"errorCode":"error.security.api-key-invalid"}'' if you use the wrong API key but the right username and password


Make sure, that in your account settings you set a different userlogin and password for the demo account. Also double check case sensitivity, and that you have everything inside ""
